api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/MessageBuilderProvider.java

```diff
@@ -0,0 +1,38 @@
+package org.eclipse.microprofile.reactive.messaging.spi;
+
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.eclipse.microprofile.reactive.messaging.MessageBuilder;
+
+import java.util.Iterator;
+import java.util.ServiceLoader;
+
+public abstract class MessageBuilderProvider {
+
+    private static volatile MessageBuilderProvider instance = null;
+
+    public abstract <T> MessageBuilder<T> newBuilder();
+
+    public static MessageBuilderProvider instance() {
+        if (instance == null) {
+            synchronized (ConfigProviderResolver.class) {
```

Comment:
I'm presuming copy/paste error, that it should be synchronized on `MessageBuilderProvider`?

Reply:
Yes thanks! I'll fix it.